### PR TITLE
PWN-7901 - Add withTld to name-service

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PW-7901

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/model/Username.kt [hunk 1]
```diff
@@ -3,6 +3,5 @@ package org.p2p.wallet.auth.model
 data class Username(
     val value: String,
     val domainPrefix: String,
-    // FIXME: New version of name service returns domain as well. We should remove domainPrefix asap
     val fullUsername: String = value + domainPrefix
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/model/Username.kt:6]: we better remove this and `domainPrefix` fields. 
All usernames will come from backend. 
And looks like we need to remove feature toggle for `username_value`
Please workaround

File: app/build.gradle.kts [hunk 2]
```diff
@@ -171,11 +171,10 @@ dependencies {
     testImplementation("io.insert-koin:koin-test:$koinVersion")
     testImplementation("io.insert-koin:koin-test-junit4:$koinVersion")
 
-    testImplementation("org.assertj:assertj-core:3.22.0")
+    testImplementation("com.willowtreeapps.assertk:assertk-jvm:0.25")
```
Comment (gslevinkov) [inline app/build.gradle.kts:174]: AssertJ erases generics when using assertThat, making assertions very bad. 
assertK has the same API

File: app/src/main/java/org/p2p/wallet/auth/model/Username.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/model/Username.kt:10]: done!
checked on main screen, search fragment

File: app/src/test/java/org/p2p/wallet/auth/username/repository/UsernameParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Test
+import org.p2p.core.utils.Constants
+import org.p2p.wallet.auth.model.Username
+import org.p2p.wallet.auth.username.repository.model.UsernameDetails
+import org.p2p.wallet.common.feature_toggles.toggles.remote.UsernameDomainFeatureToggle
+import org.p2p.wallet.utils.toBase58Instance
+
+class UsernameParserTest {
```
Comment (gslevinkov) [inline app/src/test/java/org/p2p/wallet/auth/username/repository/UsernameParserTest.kt:16]: unit test for username parsing

File: app/src/main/java/org/p2p/wallet/auth/model/Username.kt (hunk 1 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/auth/model/Username.kt:10]: I can't see, if it's done. 
You are still concatenating here: `value + domainPrefix`

File: app/src/main/java/org/p2p/wallet/auth/username/repository/UsernameParser.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.p2p.wallet.utils.Base58String
+
+/**
+ * @see [org.p2p.wallet.auth.username.repository.UsernameParserTest]
+ */
+class UsernameParser(
+    private val usernameDomainFeatureToggle: UsernameDomainFeatureToggle
+) {
+    /**
+     * @param anyUsername - username with domain prefix or not
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/username/repository/UsernameParser.kt:15]: Did you ask if we really need this logic at all? 
I don't think feature toggle is needed, moreover the domain like `.key.sol`. 

Our business couldn't own this domain as planned, so looks like the feature toggle and this logic is redundant at all

File: app/src/test/java/org/p2p/wallet/auth/username/repository/UsernameParserTest.kt [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                prop(UsernameDetails::ownerAddress).isEqualTo(ownerAddress)
+
+                val username = prop(UsernameDetails::username)
+                username.prop(Username::value).isEqualTo(expectedUsername)
+                username.prop(Username::domainPrefix).isEqualTo(expectedDomain)
+                username.prop(Username::fullUsername).isEqualTo(expectedUsername + expectedDomain)
+            }
+        }
+    }
+}
```
Comment (1sakov) [inline app/src/test/java/org/p2p/wallet/auth/username/repository/UsernameParserTest.kt:85]: looks nice!

**Review by 1sakov — APPROVED** [review]
minor comments

**Pull request merged**